### ISSUE TITLE
Questions about training poly-encoder with personachat dataset

### ISSUE THREAD

**Issue by min942773** [facebookresearch/ParlAI#3197]
Hi, I wanted to train pre-trained poly-encoder with personachat dataset that I changed a little bit.

At, first I wanted to see evaluation result so I changed the dataset and executed eval_model.py
```
parlai eval_model -t personachat:self_revised -mf zoo:pretrained_transformers/model_poly/model --eval-candidates inline
```
and the result was 
```
08:24:10 | Finished evaluating tasks ['personachat:self_revised'] using datatype valid
    accuracy  \
       .7790   
    bleu-4  \
     .7790   
    ctpb  \
   199.6   
    ctps  \
    1705   
    exps  \
   8.543   
    exs  \
   7801   
      f1  \
   .8038   
    gpu_mem  \
     .08759   
    hits@1  \
     .7790   
    hits@10  \
      .9924   
    hits@100  \
           1   
    hits@5  \
     .9677   
    loss  \
   .7951   
      lr  \
   5e-09   
    ltpb  \
   14.35   
    ltps  \
   122.6   
     mrr  \
   .8601   
    rank  \
   1.566   
     tpb  \
   213.9   
    tps  
   1828
```

And then, I trained pre-trained poly-encoder model with changed personachat data.
```
parlai train_model -t personachat:self_revised -mf zoo:pretrained_transformers/model_poly/model --eval-candidates inline -eps 1 -sval True -veps 1
```
And this is the result
```
04:18:07 | eval completed in 266.48s
04:18:07 | valid:
    accuracy  \
       .7790   
    bleu-4  \
     .7790   
    ctpb  \
   199.6   
    ctps  \
    5843   
    exps  \
   29.28   
    exs  \
   7801   
      f1  \
   .8038   
    gpu_mem  \
      .1801   
    hits@1  \
     .7790   
    hits@10  \
      .9924   
    hits@100  \
           1   
    hits@5  \
     .9677   
    loss  \
   .7951   
      lr  \
   5e-05   
    ltpb  \
   14.35   
    ltps  \
   420.2   
     mrr  \
   .8601   
    rank  \
   1.566   
    total_train_updates  \
                  65719   
     tpb  \
   213.9   
    tps  
   6263

04:18:07 | creating task(s): personachat:self_revised
04:18:07 | loading fbdialog data: /usr/local/lib/python3.6/dist-packages/data/Persona-Chat/personachat/test_self_revised.txt
04:18:07 | running eval: test
04:22:19 | eval completed in 252.37s
04:22:19 | test:
    accuracy  \
       .7817   
    bleu-4  \
     .7815   
    ctpb  \
     179   
    ctps  \
    5328   
    exps  \
   29.77   
    exs  \
   7512   
      f1  \
   .8074   
    gpu_mem  \
      .1801   
    hits@1  \
     .7817   
    hits@10  \
      .9949   
    hits@100  \
           1   
    hits@5  \
     .9704   
    loss  \
   .7977   
      lr  \
   5e-05   
    ltpb  \
   14.24   
    ltps  \
   423.9   
     mrr  \
   .8627   
    rank  \
   1.536   
    total_train_updates  \
                  65719   
     tpb  \
   193.2   
    tps  
   5752
```

As you can see, the accuracy of validation is the same.
I think poly-encoder model is not updated.

Why isn't it updating ??
How can I train this one?

**Comment by stephenroller** [comment]
You have `-eps 1.0`, or `--num-epochs 1.0`. Due to this, model will only train for one epoch, and then stop. When you try training again, it's already completed an epoch, so it doesn't continue.

Increase this value to continue training.

**Comment by stephenroller** [comment]
Just out of curious, what enviornment are you using? It looks like the log output is messing up, which is creating a suboptimal experience for you. I'd like to fix that.

**Comment by min942773** [comment]
> Just out of curious, what enviornment are you using? It looks like the log output is messing up, which is creating a suboptimal experience for you. I'd like to fix that.

it's google colab.
Thank you for your response.

**Comment by stephenroller** [comment]
You can do
```
import os
os.environ['PARLAI_FORCE_WIDTH'] = '127' # intentionally a string
```
And see if that makes things render nicer.
